core/src/main/java/org/mule/runtime/core/time/TimeSupplier.java

```diff
@@ -18,6 +22,20 @@
 {
 
     /**
+     * Returns the default instance from the mule registry
+     *
+     * @param muleContext the current {@link MuleContext}
+     * @return the default instance
+     */
+    public static TimeSupplier getDefault(MuleContext muleContext)
```

Comment:
I don't think you need this. You can inject the time supplier using AttributeDefinition.fromReferenceObject(Class<?> referenceObjectType): fromReferenceObject(TimeSupplier.class)